src/utils.ts

```diff
@@ -133,33 +134,329 @@ export namespace Utils {
+   * @param {string} networkPassphrase The network passphrase, e.g.: 'Test SDF Network ; September 2015'.
+   * @param {string[]} accountIDs The signers public keys. This list should contain the public keys for all signers that have signed the transaction.
+   * @returns {string[]} The list of signers public keys that have signed the transaction, excluding the server account ID.
+   * @example
+   *
+   * import { Networks, Transaction, Utils }  from 'stellar-sdk';
+   *
+   * const serverKP = Keypair.random();
+   * const clientKP1 = Keypair.random();
+   * const clientKP2 = Keypair.random();
+   *
+   * // Challenge, possibly built in the server side
+   * const challenge = Utils.buildChallengeTx(
+   *   serverKP,
+   *   clientKP1.publicKey(),
+   *   "SDF",
+   *   300,
+   *   Networks.TESTNET
+   * );
+   *
+   * // clock.tick(200);  // Simulates a 200 ms delay when communicating from server to client
+   *
+   * // Transaction gathered from a challenge, possibly from the client side
+   * const transaction = new Transaction(challenge, Networks.TESTNET);
+   * transaction.sign(clientKP1, clientKP2);
+   * const signedChallenge = transaction
+   *         .toEnvelope()
+   *         .toXDR("base64")
+   *         .toString();
+   *
+   * // The result below should be equal to [clientKP1.publicKey(), clientKP2.publicKey()]
+   * Utils.verifyChallengeTxSigners(signedChallenge, serverKP.publicKey(), Networks.TESTNET, threshold, clientKP1.publicKey(), clientKP2.publicKey());
+   */
+  export function verifyChallengeTxSigners(
+    challengeTx: string,
+    serverAccountID: string,
+    networkPassphrase: string,
+    accountIDs: string[],
+  ): string[] {
+    // Read the transaction which validates its structure.
+    const { tx } = readChallengeTx(
+      challengeTx,
+      serverAccountID,
+      networkPassphrase,
+    );
+
+    // Ensure the server account ID is an address and not a seed.
+    let serverKP: Keypair;
+    try {
+      serverKP = Keypair.fromPublicKey(serverAccountID); // can throw 'Invalid Stellar public key'
+    } catch (err) {
+      throw new Error(
+        "Couldn't infer keypair from the provided 'serverAccountID': " +
+          err.message,
+      );
+    }
+
+    // Deduplicate the client signers and ensure the server is not included
+    // anywhere we check or output the list of signers.
+    const clientSigners = new Set<string>();
+    for (const signer of accountIDs) {
+      // Ignore the server signer if it is in the signers list. It's
+      // important when verifying signers of a challenge transaction that we
+      // only verify and return client signers. If an account has the server
+      // as a signer the server should not play a part in the authentication
+      // of the client.
+      if (signer === serverKP.publicKey()) {
+        continue;
+      }
+
+      // Ignore non-G... account/address signers.
+      if (!StrKey.isValidEd25519PublicKey(signer)) {
```

Comment:
Hey @leighmcculloch, please check if this is what you had in mind: https://github.com/stellar/js-stellar-sdk/pull/497/commits/3b59cb56b59be6774f4aa9aa6d2c566ff2849721

Since the JS SDK doesn't have a method like the Go SDK `strkey.Version`, I'm manually checking for a leading `'G'` character in the public key.

Other than that, I believe the JS SDK is in the same page with the Go SDK.